momento-sdk/src/main/java/momento/sdk/messages/CacheSortedSetFetchResponse.java

```diff
@@ -25,54 +23,17 @@ public Hit(List<_SortedSetElement> elements) {
       this.elements = elements;
     }
 
-    public NavigableSet<ScoredElement<String>> elementsSet() {
-      return elements.stream()
-          .map(e -> new ScoredElement<>(e.getValue().toStringUtf8(), e.getScore()))
-          .collect(Collectors.toCollection(TreeSet::new));
-    }
-
-    public List<ScoredElement<String>> elementsList() {
-      return elements.stream()
-          .map(e -> new ScoredElement<>(e.getValue().toStringUtf8(), e.getScore()))
-          .collect(Collectors.toList());
-    }
-
-    public List<ScoredElement<byte[]>> elementsByteArrayList() {
-      return elements.stream()
-          .map(e -> new ScoredElement<>(e.getValue().toByteArray(), e.getScore()))
-          .collect(Collectors.toList());
-    }
-
-    /**
-     * Gets a map of the retrieved elements as UTF-8 strings to their scores. The order of the map
-     * is determined by the options used in the method call.
-     *
-     * @return An ordered map of String elements to scores
-     */
-    public Map<String, Double> elementsMap() {
-      return elements.stream()
-          .collect(
-              Collectors.toMap(
-                  e -> e.getValue().toStringUtf8(),
-                  _SortedSetElement::getScore,
-                  (u, v) -> u,
-                  LinkedHashMap::new));
-    }
-
     /**
-     * Gets a map of the retrieved elements as byte arrays to their scores. The order of the map is
-     * determined by the options used in the method call.
+     * Gets a navigable set of the retrieved elements and their scores. The set is in ascending
+     * order by default and can be viewed in descending order with {@link
+     * NavigableSet#descendingSet()}.
      *
-     * @return An ordered map of byte[] elements to scores
+     * @return An ordered set of elements to scores
      */
-    public Map<byte[], Double> elementsByteArrayMap() {
+    public NavigableSet<ScoredElement> elementsSet() {
```

Comment:
still curious to know if someone could pass this back in as input.  I'm not reviewing this PR as thoroughly as I normally would because it's late and we're pre-1.0 and I don't want to keep you waiting, so not a blocker, but food for thought :)

Reply:
Right now you could not. I can update ScoredElement to be more usable by consumers. This opens a more broad question about encapsulating classes as a solution to the type erasure problem, since ScoredElement effectively is such a wrapping class.

Reply:
let's keep an eye out for this in bug bash.  I think we should assign owners to different collection types to try to get coverage.  you should remind me to choose sortedset (possibly along with some other folks if we have enough volunteers).